cli/src/cli.rs

```diff
@@ -0,0 +1,64 @@
+use clap::{crate_authors, crate_version, Parser};
+use color_eyre::{eyre::eyre, Report, Result};
+use std::path::PathBuf;
+
+use crate::config::{Config, FromFile, LoggerConfig};
+const VERSION: &str = concat!(
+    "BOB-GUI VERSION: ",
+    crate_version!(),
+    "\n",
+    "BUILT AT: ",
+    env!("BUILD_TIME"),
+    "\n",
+    "COMMIT HASH: ",
+    env!("GIT_HASH"),
+    "\n",
+    "GIT BRANCH: ",
+    env!("GIT_BRANCH"),
+    "\n",
+    "GIT TAG: ",
```

Comment:
Branch and tag should be combined, because they are mutually exclusive. You can pass the value from `${{ github.ref }}`

Reply:
Also change the name of the env variable: it should start with the application name (e.g. `MANGUI_BUILD_BRANCH_TAG`)

Reply:
Also use `option_env!(..).unwrap_or("-")` for branch/tag name and for commit hash, because these parameters are optional and only available when building with CI